### PR TITLE
Upgrade to new rhetos

### PR DESCRIPTION
**Pull request by ibarban**

### CODE REVIEW

File: Plugins/AdminSetup/AdminSetup.csproj [hunk 1]
```diff
@@ -228,11 +63,24 @@
...
   </ItemGroup>
+  <ItemGroup>
+    <PackageReference Include="Microsoft.AspNet.WebPages">
+      <Version>2.0.20710</Version>
+    </PackageReference>
+    <PackageReference Include="Microsoft.AspNet.WebPages.WebData">
+      <Version>2.0.20710</Version>
+    </PackageReference>
+    <PackageReference Include="Rhetos">
+      <Version>3.1.0-dev*</Version>
```
Comment (bantolov) [inline Plugins/AdminSetup/AdminSetup.csproj]: It the star in `3.1.0-dev*` required? Is it standard for references in csproj?

Comment (bantolov) [inline Plugins/AdminSetup/AdminSetup.csproj]: We should update Rhetos version to 4.0.0.

File: Plugins/InitAspNetDatabase/Program.cs [hunk 2]
```diff
@@ -37,7 +36,10 @@ static int Main(string[] args)
             string errorMessage = null;
             try
             {
-                Paths.InitializeRhetosServerRootPath(Path.Combine(AppDomain.CurrentDomain.BaseDirectory, @"..\.."));
+                var rootPath = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, @"..\..");
+                var configurationProvider = new Rhetos.ConfigurationBuilder().
+                    AddRhetosAppConfiguration(rootPath).Build();
```
Comment (sstublic) [inline Plugins/InitAspNetDatabase/Program.cs]: Code style: redundant `Rhetos` namespace qualifier, line breaks. Should read something like this:
```
            var configurationProvider = new ConfigurationBuilder()
                .AddRhetosAppConfiguration(rootPath)
                .Build();
```

File: Plugins/AdminSetup/Program.cs [hunk 3]
```diff
@@ -51,7 +45,10 @@ static int Main(string[] args)
             string errorMessage = null;
             try
             {
-                Paths.InitializeRhetosServerRootPath(Path.Combine(AppDomain.CurrentDomain.BaseDirectory, @"..\.."));
+                var configurationProvider = new Rhetos.ConfigurationBuilder().
+                    AddRhetosAppConfiguration(GetRhetosRootFolder()).Build();
+                LegacyUtilities.Initialize(configurationProvider);
```
Comment (sstublic) [inline Plugins/AdminSetup/Program.cs]: Intention is not clear to me here. Probably LegacyUtilities are needed even if try..catch fails? 

However, this configuration is redundant and slightly incompatible (missing `AddConfigurationManagerConfiguration`) with the one created at the start of try...catch block. Maybe create container at the top code level to clearly display when it happens as it also initializes LegacyUtilities implicitly. 

If container creation fails itself, I'm not sure if the code is expected to continue at all?

**Review by bantolov — CHANGES_REQUESTED** [review]
Just changing the PR status.

File: Plugins/AdminSetup/Program.cs (hunk 3 above)
Comment (bantolov) [reply Plugins/AdminSetup/Program.cs]: This allows call to SetUpAdminAccount to work even if CreateAdminUserAndPermissions failed. It is a strange feature, but we'll leave it for now.
We have modified the code and added comment to make the intention more clear.

**Review by bantolov — APPROVED** [review]

**Pull request merged**